packages/api/src/Base.ts

```diff
@@ -515,24 +515,30 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
       ) as C;
 
     // FIXME The unknown cast is needed since the onCall result, `C | S` cannot
-    // be converted from C to the actual result required
+    // be converted from C to the actual result required (same with multi and size)
     decorated.hash = (arg?: CodecArg): HashResult<C, S> =>
       onCall(
         (arg: CodecArg) => this._rpcRx.state.getStorageHash([method, arg]),
         [arg]
       ) as unknown as HashResult<C, S>;
 
-    // FIXME as above...
+    decorated.key = (arg?: CodecArg): string =>
+      u8aToHex(compactStripLength(method(arg))[1]);
+
+    decorated.multi = (args: Array<CodecArg>, callback: CodecArrayCallback): M =>
+      onCall(
+        () => this._rpcRx.state.subscribeStorage(args.map((arg) => [method, arg])),
+        [],
+        callback
+      ) as unknown as M;
```

Comment:
Instead of introducing M, this could be a C imo. And `type RxArrayResult = Observable<Vector<Codec>>;
`

Reply:
The issue with Vector... it is not quite a generic "take in anything" approach, it assumes exactly the same items in the vector, hence taking in a type. A `Tuple` on principle can have anything, but has the same "tell me what will be in it at construction". 

So even just trying to squeeze in something like that [here](https://github.com/polkadot-js/api/blob/master/packages/rpc-core/src/index.ts#L235) creates some ripple-effects down the line. (Caching, pulling older values from existing arrays since not everything is specified, etc.)